### PR TITLE
MT-12-Topic-4-HW

### PR DESCRIPTION
**Pull request by AliakseiDziardziyashenka**
Created five projects in my MentoringHomeWork folder

### CODE REVIEW

File: MentoAD.sln [hunk 1]
```diff
@@ -9,6 +9,19 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HW03.Calculator.Static", "M
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HW03.Birthday", "MentoringHomeWork\HW3.0\HW03.Birthday\HW03.Birthday.csproj", "{6E851AD7-F6F1-4DBF-BEBB-D7922E194A3A}"
 EndProject
+<<<<<<< HEAD
```
Comment (batonkarton) [inline MentoAD.sln]: You tried to resolve merge conflicts here using UI probably - so these comments created an issue for you
image

File: MentoringHomeWork/HW4/Task2/sum/sum.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+public class Sum
+{
+    public void Task2()
+    {
+        var result = 0;
+        Getnumber();
+        CheckSum();
+        return;
+
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task2/sum/sum.cs]: Be careful with new lines - minor: delete useless spaces, there should be just one

File: MentoringHomeWork/HW4/Task2/sum/sum.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+namespace Task2.sum;
+
+public class Sum
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task2/sum/sum.cs]: The name of the class is Sum but the name of the file name is sum.cs - please remember: class name and file name usually are the same and in PascalCase. Refactor please in all places below.

Comment (batonkarton) [reply MentoringHomeWork/HW4/Task2/sum/sum.cs]: I would say that the name of the class "Sum" is an operation - not goal of this class. 
You should think in terms of what this class does—validate smth: Validator, help with smth: Helper, model of smth: BookModel, and so on. It should be something like a Calculator. Please rename them.

File: MentoringHomeWork/HW4/Task1/Calculate/Calculate.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+namespace Task1.Calculate;
+
+public class Calculate
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task1/Calculate/Calculate.cs]: You should think in terms of what this class does—validate smth: Validator, help with smth: Helper, model of smth: BookModel, and so on.
It should be something like a Calculator. Please rename.

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+public class Sum3
+{
+    public void Task3()
+    {
+        var result = 0;
+        Getnumber();
+        CheckSum();
+        return;
+
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: Be careful with new lines - minor: delete useless spaces, there should be just one

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                }
+                else if (resultSum > result)
+                {
+                    Console.WriteLine("There should be less");
+                    Task3();
+                }
+                else if (result != resultSum)
+                {
+                    Console.WriteLine("Incorrect, please try again.");
+                    Task3(); // Рекурсивный вызов при неправильном ответе
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: Remove comments

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+                }
+                else
+                {
+                    Console.WriteLine("Correct! The result is: " + result);
+                }
+            }
+            else
+            {
+                Console.WriteLine("Error, incorrect input for the sum. Please try again.");
+                Task3(); // Рекурсивный вызов при некорректном вводе
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: Remove comments

File: MentoringHomeWork/HW4/Task6/the player's movement/TheDirection.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+public class TheDirection
+{
+    public void Main()
+    {
+        Console.WriteLine("Welcome ,lets start snowboarding \ud83c\udfc2");
+        Console.WriteLine("Just write finish if you want exit");
+        Console.WriteLine("Choose your direction(use 'W,A,S,D')");
+
+        Dirextion();
+        void Dirextion()
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task6/the player's movement/TheDirection.cs:12]: I believe it's not your code - let's discuss it. Leave it like it is now.

File: MentoringHomeWork/HW4/Task4/sum4/sum4.cs [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        var number = 0;
+        var result = 0;
+        var number2 = 0;
+
+        Getnumber();
+        GetOperator();
+        CheckSum();
+        return;
+
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task4/sum4/sum4.cs]: Be careful with new lines - minor: delete useless spaces, there should be just one

File: MentoringHomeWork/HW4/Task2/sum/sum.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+            }
+            else
+            {
+                Console.WriteLine("Error, incorrect input for the first number. Please try again.");
+                Task2(); 
+            }
+        }
+
+        void CheckSum()
+        {
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task2/sum/sum.cs]: image
is it what you expect?

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                }
+            }
+            else
+            {
+                Console.WriteLine("Error, incorrect input for the first number. Please try again.");
+                Task3();
+            }
+        }
+
+        void CheckSum()
```
Comment (batonkarton) [inline MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs:39]: image
minor - not required change

**Review by batonkarton — CHANGES_REQUESTED** [review]
Please make fixes according comments

**Comment by batonkarton** [comment]
@AliakseiDziardziyashenka reminder about commits messages:
_**Commits convention:
Convention for commit message: {Ticket-Number}:{your description}
ex. MT-7: added review fixes**_

https://mentoring-itra-ak.atlassian.net/wiki/spaces/MAD/pages/2326530/Mentoring+GIT+Flow

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs (hunk 11 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs:39]: resolved

File: MentoringHomeWork/HW4/Task2/sum/sum.cs (hunk 10 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task2/sum/sum.cs]: I dont have this issue

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs (hunk 5 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: resolved

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs (hunk 6 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: resolved

File: MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs (hunk 7 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task3/Sum3/Sum3.cs]: resolved

File: MentoringHomeWork/HW4/Task4/sum4/sum4.cs (hunk 9 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task4/sum4/sum4.cs]: resolved

File: MentoringHomeWork/HW4/Task2/sum/sum.cs (hunk 2 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW4/Task2/sum/sum.cs]: resolved

**Review by batonkarton — APPROVED** [review]

**Pull request merged**